internal/fleet/agent_policy/models.go

```diff
@@ -100,5 +179,47 @@ func (model agentPolicyModel) toAPIUpdateModel() kbapi.PutFleetAgentPoliciesAgen
 		Namespace:          model.Namespace.ValueString(),
 	}
 
-	return body
+	if len(model.GlobalDataTags.Elements()) > 0 {
```

Comment:
It looks like we copy this between create/update. Can we pull it out and DRY this code up a little?